### PR TITLE
Add build_rpm script and rpm spec

### PR DESCRIPTION
**Pull request by romange**
Also, link dragonfly with stdlib++ and libgcc statically.

### CODE REVIEW

**Comment by romange** [comment]
Tested here: https://github.com/dragonflydb/dragonfly/actions/runs/6133194672 
this failed because our regression tests are failing.

File: .github/workflows/release.yml [hunk 1]
```diff
@@ -96,37 +97,61 @@ jobs:
...
+      image: ghcr.io/romange/${{ matrix.container }}
     steps:
     - uses: actions/checkout@v3
       with:
         submodules: true
     - name: Configure
       run: |
-          apt update && apt install -y debhelper moreutils pip jq
+          if [ -f /etc/redhat-release ]; then
+            dnf install -y rpm-build glibc-static libstdc++-static
```
Comment (royjacobson) [inline .github/workflows/release.yml:117]: Why do we need `glibc-static`? I thought we're only changing libgcc and libstdc++.

**Comment by royjacobson** [comment]
I get the hate for dynamically linking external libraries, but `libstdc++` is generally forward compatible, no? I don't want to increase binary sizes without a good reason.

**Comment by romange** [comment]
libstd++ is not compatible between distributions, imho. you can not build
on debian and run the binary on centos. Btw, you are welcome to check it -
you can build the release from main and try running it on fedora container.

gcc-static is what allows me to link with libgcc. this package is separate
on fedora.

I think that building a bigger binary will be offsest by using a slimmer
container(ubuntu-prod).

**Comment by chakaz** [comment]
I _think_ that libstdc++ is only ABI compatible if it's built using the same macros, and it's definitely not guaranteed that the same macros are used across machines / distros...

**Comment by royjacobson** [comment]
> libstd++ is not compatible between distributions, imho. you can not build on debian and run the binary on centos. Btw, you are welcome to check it - you can build the release from main and try running it on fedora container.

It is forward compatible but not backwards compatible. So you can link (and compile) against an old version and then load a newer version dynamically. In fedora we have GCC 9 from 4 years ago so it makes sense that dragonfly doesn't run there. (@chakaz is right that you can configure libstdc++ in ways that would introduce incompatibility but I don't think it's something that is likely).
If the argument is we want to use a newer `libstdc++` even on older distributions - fine, let's compile statically with GCC13. But I am not sure that linking `libstdc++` statically would give us less deployment friction.

> gcc-static is what allows me to link with libgcc. this package is separate on fedora. I think that building a bigger binary will be offsest by using a slimmer container(ubuntu-prod).

This PR adds `glibc-static`, not `gcc-static`.

**Comment by chakaz** [comment]
> But I am not sure that linking libstdc++ statically would give us less deployment friction.

What makes you say that? Won't it save the DLL hell equivalent of whatever libstdc++ is installed on old systems, etc?

File: .github/workflows/release.yml (hunk 1 above)
Comment (romange) [reply .github/workflows/release.yml:117]: will remove

**Comment by romange** [comment]
Some data point from fedora-30 distribution:
shared libs (stdlib++  and gcc) : 2179528 bytes
when linking statically, our binary will grow by 1534456 bytes.
That means our docker images will actually become smaller.

**Comment by romange** [comment]
I actually prefer compiling it statically on newer distributions if it allows us running dragonfly on the old ones but I think we will also have problems with glibc library.

**Comment by royjacobson** [comment]
> I actually prefer compiling it statically on newer distributions if it allows us running dragonfly on the old ones but I think we will also have problems with glibc library.

Yeah I guess glibc might be a problem. Still, I don't want to distribute binaries built with GCC9 :face_exhaling:

**Comment by romange** [comment]
We do not yet release it, it's a first sep. I believe once submitted and if
it bothers you too much you will find a way how to install gcc of your
choice on fedora 30 :)

**Pull request merged**